goGene/goGene.go

```diff
@@ -0,0 +1,176 @@
+package goGene
+
+import (
+	"errors"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/vcf"
+)
+
+type Feature int32
+type MutationType byte
+
+const (
+	Intron   Feature = -1
+	UtrThree Feature = -3
+	UtrFive  Feature = -5
+	// All positive values refer to cDNA position
+
+	Silent       MutationType = 0
+	Missense     MutationType = 1
+	Nonsense     MutationType = 2
+	Frameshift   MutationType = 3
+	Intergenic   MutationType = 4
+	Intronic     MutationType = 5
+	Splice       MutationType = 6 // +/- 1-2 of E-I boundary
+	FarSplice    MutationType = 7 // +/- 3-10 of E-I boundary
+	DisruptStart MutationType = 8
+	DisruptStop  MutationType = 9
+)
+
+// GoGene is a processed version of a gtf record that enables easy
+// traversal and manipulation of genes on the genomic and mRNA levels.
+type GoGene struct {
```

Comment:
I think these fields might each need a sentence of documentation.  I am trying to imagine writing code to turn data from a file in genePred into a GoGene, and I don't think I would always know what to do.
What about Gene instead of GoGene?  We have been using the Go prefix to mean a function that will launch its own thread.  Or EasyGene, InternalGene?